libs/utils/sidekick.js

```diff
@@ -22,17 +27,37 @@ export default function init({ createTag, loadBlock, loadScript, loadStyle }) {
     getModal(null, { id: 'preflight', content, closeEvent: 'closeModal' });
   };
 
+  const addVersion = async () => {
+    const sk = document.querySelector('helix-sidekick');
+    const statusJson = JSON.parse(sk.getAttribute('status'));
+    const sourceUrl = statusJson?.edit?.url;
+    const sourceCode = sourceUrl?.match(/sourcedoc=([^&]+)/)[1];
+    const sourceId = decodeURIComponent(sourceCode);
+    const url = `https://adobe.sharepoint.com/sites/adobecom/_api/web/GetFileById('${sourceId}')`;
+
+    const options = getReqOptions({
+      method: 'POST',
+      accept: 'application/json; odata=nometadata',
+      contentType: 'application/json;odata=verbose',
+    });
+    await fetch(`${url}/Publish('Last Published version')`, { ...options, keepalive: true });
+  };
+
   // Support for legacy manifest v2 - Delete once everyone is migrated to v3
   document.addEventListener('send-to-caas', async (e) => {
     const { host, project, branch, repo, owner } = e.detail;
     const { sendToCaaS } = await import('../../tools/send-to-caas/send-to-caas.js');
     sendToCaaS({ host, project, branch, repo, owner }, loadScript, loadStyle);
   });
 
+  // fetch sharepoint access token
+  loginToSharePoint(['https://adobe.sharepoint.com/.default']);
```

Comment:
Is this really needed at this point? Looks like you're already calling `loginToSharePoint` in `version-history/view.js`.

Reply:
@honstar This is a separate flow in which the version history view.js isn't rendered. This is to create a version when there is a publish event on the file. So, yes this is needed.

Reply:
Ok, I see what you mean. I assume this doesn't affect an author as long as he/she is already logged into sharepoint, right?

Reply:
Yes. It wouldn't.